pilot/pkg/networking/plugin/authz/validate.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package authz
+
+import (
+	"fmt"
+	"strings"
+
+	rbacproto "istio.io/api/rbac/v1alpha1"
+)
+
+// validateRuleForTCPFilter checks if the given rule is good for RBAC TCP filter. Returns nil if the
+// rule doesn't include any HTTP specifications.
+func validateRuleForTCPFilter(rule *rbacproto.AccessRule) error {
+	if rule == nil {
+		return nil
+	}
+
+	if len(rule.Paths) != 0 {
+		return fmt.Errorf("paths(%v) not supported", rule.Paths)
+	}
+	if len(rule.Methods) != 0 {
+		return fmt.Errorf("methods(%v) not supported", rule.Methods)
+	}
+	for _, constraint := range rule.Constraints {
+		if strings.HasPrefix(constraint.Key, attrRequestHeader) {
+			return fmt.Errorf("constraint(%v) not supported", constraint)
+		}
+	}
+
+	return nil
+}
+
+// validateBindingsForTCPFilter checks if the give binding is good for RBAC TCP filter. Returns nil
+// if the binding doesn't include any HTTP specifications.
+func validateBindingsForTCPFilter(bindings []*rbacproto.ServiceRoleBinding) error {
+	for _, binding := range bindings {
+		for _, subject := range binding.Subjects {
+			if subject.Group != "" {
+				return fmt.Errorf("group(%v) not supported", subject.Group)
+			}
+			for k := range subject.Properties {
+				switch k {
+				case attrSrcIP:
+					// TODO(yangminzhu): Support attrSrcNamespace and attrSrcPrincipal once
+					// https://github.com/envoyproxy/envoy/pull/4250 is merged.
```

Comment:
shouldn't IP be part of the attributes available to envoy already?
Also, are you going to interpret principal_user (a string) as IP? Even then, namespace won't work.

Reply:
Yes, it supports only the IP property by returning a nil error, any other property other than `attrSrcIP` will result an error.
For `attrSrcPrincipal` and `attrSrcNamespace`, I'm planning to check it directly in the peer's certificate using regex once the Envoy PR is merged. In Istio, I think we encode the source namespace into the certificate identity.